engine/pom.xml

```diff
@@ -273,39 +273,6 @@
         </plugins>
     </build>
     <profiles>
-        <profile>
```

Comment:
We can also clean up the parent pom as we used to require the `java.xml.bind` module (just search for it in the parent pom and you'll see the start of the clean up to do).

I don't think it's worth keeping the existing add-modules machinery as I don't think we will need it anymore.

We can also clean up the `-J--add-modules=java.xml.ws.annotation` compilation parameter while at it. We won't have any `@Generated` annotation anymore.